indra/llwindow/llkeyboard.cpp

```diff
@@ -254,6 +257,10 @@ BOOL LLKeyboard::handleTranslatedKeyDown(KEY translated_key, U32 translated_mask
 	mKeyDown[translated_key] = TRUE;
 	mCurTranslatedKey = (KEY)translated_key;
 	handled = mCallbacks->handleTranslatedKeyDown(translated_key, translated_mask, repeated);
+    if (!handled && LLGameControl::getIncludeKeyboardButtons())
+    {
+        LLGameControl::onButton(translated_key, true);
+    }
```

Comment:
Indentation is bad here, but there are plans to eventually convert all of the viewer code over to four-spaces instead of tabs.  Maybe that work will get done before this is shipped, so I'll leave this alone for now.

The more interesting aspect of these specific changes: we treat all "unhandled" keystrokes as game controller input (if enabled).  This will allow LSL scripts on attachments and seats to have access to keyboard control no permissions necessary (e.g. leave the feature disabled if you don't want to broadcast).